ironfish-cli/src/commands/chain/foo.ts

```diff
@@ -0,0 +1,131 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import {
+  Assert,
+  Block,
+  GENESIS_BLOCK_SEQUENCE,
+  Meter,
+  NodeUtils,
+  Spend,
+  TimeUtils,
+} from '@ironfish/sdk'
+import { CliUx, Flags } from '@oclif/core'
+import { IronfishCommand } from '../../command'
+import { LocalFlags } from '../../flags'
+import { ProgressBar } from '../../types'
+
+export default class Repair extends IronfishCommand {
+  static description = ''
+
+  static flags = {
+    ...LocalFlags,
+    dry: Flags.boolean({
+      default: false,
+      description: 'Dry run repair first',
+    }),
+    batchSize: Flags.integer({
+      default: 1000,
+      description: 'How many blocks to load in parallel',
+    }),
+    start: Flags.integer({
+      default: GENESIS_BLOCK_SEQUENCE,
+      description: 'Where to start repairing',
+    }),
+  }
+
+  async start(): Promise<void> {
+    const { flags } = await this.parse(Repair)
+
+    const node = await this.sdk.node()
+    await NodeUtils.waitForOpen(node)
+
+    let nullifierTreeIndex = 0
+    let nullifierRepaired = 0
+    let current = flags.start
+    const stop = node.chain.head.sequence
+
+    const processBatch = async (blocks: Block[]): Promise<void> => {
+      for (const block of blocks) {
+        for (const spend of block.spends()) {
+          const contains = await node.chain.nullifiers.contains(spend.nullifier)
+
+          if (!contains) {
+            nullifierRepaired++
+
+            this.log(
+              `\rMissing nullifier: block ${
+                block.header.sequence
+              }, index: ${nullifierTreeIndex}: ${spend.nullifier.toString(
+                'hex',
+              )} (repaired ${nullifierRepaired})${''.padEnd(10, ' ')}`,
+            )
+
+            refreshProgressBar()
+
+            if (!flags.dry) {
+              const merkleHash = node.chain.nullifiers.hasher.merkleHash(spend.nullifier)
```

Comment:
Was thinking if we should double check whether the leaf at that index actually corresponds to that nullifier. Can't think of why it wouldn't but couldn't hurt to check since we are updating the index position and want it to be 100% correct. Also this would be rare so wouldn't really add much time to the run
```
const existingNullifier = await node.chain.nullifiers.leaves.get(nullifierTreeIndex)
```

Reply:
Sure, added.